src/app/components/JumpTo/index.tsx

```diff
@@ -0,0 +1,74 @@
+/** @jsx jsx */
+import { useContext } from 'react';
+import { jsx } from '@emotion/react';
+import { ServiceContext } from '#contexts/ServiceContext';
+import useViewTracker from '#app/hooks/useViewTracker';
+import useClickTrackerHandler from '#app/hooks/useClickTrackerHandler';
+import { EventTrackingMetadata } from '#app/models/types/eventTracking';
+import Text from '#app/components/Text';
+import InlineLink from '#app/components/InlineLink';
+
+interface JumpToHeading {
+  id: string;
+  title: string;
+}
+
+interface JumpToProps {
+  jumpToData: {
+    model: {
+      jumpToHeadings: Array<{ heading: string }>;
+    };
+  };
+  eventTrackingData?: EventTrackingMetadata;
+}
+
+const JumpTo = ({ jumpToData, eventTrackingData }: JumpToProps) => {
+  const { translations } = useContext(ServiceContext);
+  const { jumpTo = 'Jump to' } = translations?.articlePage || {};
+
+  const viewRef = useViewTracker(eventTrackingData);
+  const clickTrackerHandler = useClickTrackerHandler({
+    ...eventTrackingData,
+    identifier: 'JumpTo',
+  });
+
+  const subheadlines: JumpToHeading[] = jumpToData?.model?.jumpToHeadings.map(
+    (item, index) => ({
+      id: `jump-to-${index}`,
+      title: item.heading,
+    }),
+  );
```

Comment:
Don't think we need to do any additional mapping here.

The data structure passed in when its on an Article page should just allow us to map over the elements on https://github.com/bbc/simorgh/pull/12123/files#diff-003060b67ed07ff784c6e0dc683581c56042fd0fcadfc71c20458576c0eb94b9R58

Reply:
E.g.:

```
{subheadlines.map(({ heading }, index) => (
````

Reply:
Basically by the time the component renders on the ArticlePage, we'll have access to all the `model` contents in the response. So we can just access `jumpToHeadings` directly from the component props:

![](https://private-user-images.githubusercontent.com/120198886/382292119-0d77b5fb-8c63-4410-941c-2994eea0ce85.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA1NDU0MzksIm5iZiI6MTczMDU0NTEzOSwicGF0aCI6Ii8xMjAxOTg4ODYvMzgyMjkyMTE5LTBkNzdiNWZiLThjNjMtNDQxMC05NDFjLTI5OTRlZWEwY2U4NS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTAyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEwMlQxMDU4NTlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05OTdmMDU2ZjNlZTZkOTViMmIxYTI1MzQ0YjZmYWVjMTMzMTk2MDU1YjY0YTAwZmE5YzZkMDRkZGU3MzcyMmZlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9._ng6iOypnMOkXCxgYlEvEvkS0FJrkfigUe76uVeYSo0)

Reply:
Updated now - just wondering if the `sanitisedId` is also needed on the `data-test-id`?

Reply:
Yea probably should be used for `data-test-id` as well 👍

Reply:
Cool! it's already on there